packages/console/src/assets/docs/guides/m2m-general/README.mdx

```diff
@@ -1 +1,110 @@
-## Replace this with actual guide
\ No newline at end of file
+import InlineNotification from '@/ds-components/InlineNotification';
+import Steps from '@/mdx-components-v2/Steps';
+import Step from '@/mdx-components-v2/Step';
+import ApplicationCredentials from '@/mdx-components-v2/ApplicationCredentials';
+import EnableAdminAccessSrc from './assets/enable-admin-access.png';
+import AppIdentifierSrc from './assets/api-identifier.png';
+
+<Steps>
+<Step title="Intro">
+Machine to Machine (M2M) is a common practice to authenticate if you have an app that needs to directly talks to resources. E.g., an API service that updates users' custom data in Logto, a statistic service that pulls daily orders, etc.
+
+Usually, an M2M app doesn't need user interactions, i.e., it has no UI.
+
+</Step>
+<Step title="Locate the app ID and app secret" subtitle="1 step">
+Get your App ID and App Secret.
+
+<ApplicationCredentials id={props.app.id} secret={props.app.secret} />
```

Comment:
you can use guide context in this component. no need to pass the props

```ts
const { app: { id, secret } } = useContext(GuideContext);
```

Reply:
updated